src/gDP.cpp

```diff
@@ -792,6 +797,126 @@ void gDPSetScissor(u32 mode, s16 xh, s16 yh, s16 xl, s16 yl)
+
+		if (config.generalEmulation.enableFragmentDepthWrite == 0)
+			drawer.clearDepthBuffer();
+		else
+			depthBufferList().setCleared(true);
+
+		// Deduce ulx, uly, lrx, lry from addr, size, and colorImageStart
+		// Currently I am assuming a simple letterbox case - top and bottom are not cleared
+		const u32 lineSize = screenWidth * 2;
+		const u32 ulx = 0;
+		const u32 uly = (addr - depthImageStart) / lineSize;
+		const u32 lrx = screenWidth;
+		const u32 lry = uly + length / lineSize;
+
+		if (config.generalEmulation.enableFragmentDepthWrite != 0) {
+			// Pretend that we are drawing the rectangle over ZB with fill mode
+			BackupRestore backupColorImageAddress(gDP.colorImage.address);
+			BackupRestore backupWidth(gDP.colorImage.width);
+			const unsigned int backupDepthSource = gDP.otherMode.depthSource;
+			const unsigned int backupCycleType = gDP.otherMode.cycleType;
+
+			gDP.colorImage.address = gDP.depthImageAddress;
+			gDP.colorImage.width = screenWidth;
+			gDP.otherMode.cycleType = G_CYC_FILL;
+
+			drawer.drawRect(ulx, uly, lrx, lry);
+			frameBufferList().setBufferChanged(f32(lry));
+
+			gDP.otherMode.depthSource = backupDepthSource;
+			gDP.otherMode.cycleType = backupCycleType;
+		}
+
+		// TODO: Following the same logic and not filling in zb values in RDRAM, why?
+		return;
+	}
+
+	// Either RGBA16 or RGBA32. This heavily assume "niceness" of developers to bind color buffer before memset...
+	const u32 colorSize = gDP.colorImage.size < 3 ? 2 : 4;
+	const u32 colorImageStart = gDP.colorImage.address;
+	const u32 colorImageEnd = gDP.colorImage.address + screenWidth * screenHeight * colorSize;
+	if (colorImageStart <= addr && addr < colorImageEnd) {
+		// FB clear
+		f32 fillColor32[4];
+		getFillColor(fillColor, fillColor32);
+		gDP.rectColor.r = fillColor32[0];
+		gDP.rectColor.g = fillColor32[1];
+		gDP.rectColor.b = fillColor32[2];
+		gDP.rectColor.a = fillColor32[3];
+
+		const u32 lineSize = colorSize * screenWidth;
+		const u32 ulx = 0;
+		const u32 uly = (addr - colorImageStart) / lineSize;
+		const u32 lrx = screenWidth;
+		const u32 lry = uly + length / lineSize;
+
+		{
+			BackupRestore backupFillColor(gDP.fillColor.color);
+			const unsigned int backupDepthSource = gDP.otherMode.depthSource;
+			const unsigned int backupCycleType = gDP.otherMode.cycleType;
+
+			gDP.fillColor.color = fillColor;
+			gDP.otherMode.cycleType = G_CYC_FILL;
+
+			drawer.drawRect(ulx, uly, lrx, lry);
+			if (auto cur = frameBufferList().getCurrent())
+				cur->setBufferClearParams(gDP.fillColor.color, ulx, uly, lrx, lry);
+
+			frameBufferList().setBufferChanged(f32(lry));
+
+			gDP.otherMode.depthSource = backupDepthSource;
+			gDP.otherMode.cycleType = backupCycleType;
+		}
+
+		if (config.frameBufferEmulation.copyFromRDRAM != 0) {
+			for (int i = 0; i < length / 4; i++) {
+				*(u32*)(RDRAM + addr + i * 4) = fillColor;
+			}
```

Comment:
```suggestion
			for (int i = 0; i < length; i+=4) {
				*(u32*)(RDRAM + addr + i ) = fillColor;
			}
```
It should be faster. Or
```
u32* pDest = interpret_cast<u32*>(RDRAM + addr);
length /= 4; // or length >>= 2;
for (int i = 0; i < length; i++) {
	pDest[i] = fillColor;

```
The same below.